src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/LabeledVariantAnnotationsWalker.java

```diff
@@ -0,0 +1,382 @@
+            optional = true)
+    boolean useASAnnotations = false;
+
+    @Argument(
+            fullName = IGNORE_FILTER_LONG_NAME,
+            doc = "Ignore the specified filter(s) in the input VCF.",
+            optional = true)
+    private List<String> ignoreInputFilters = new ArrayList<>();
+
+    @Argument(
+            fullName = IGNORE_ALL_FILTERS_LONG_NAME,
+            doc = "If true, ignore all filters in the input VCF.",
+            optional = true)
+    private boolean ignoreAllFilters = false;
+
+    // TODO this is a perhaps vestigial argument inherited from VQSR; its impact and necessity could be reevaluated
+    @Argument(
+            fullName = DO_NOT_TRUST_ALL_POLYMORPHIC_LONG_NAME,
+            doc = "If true, do not trust that unfiltered records in the resources contain only polymorphic sites. " +
+                    "This may increase runtime.",
+            optional = true)
+    private boolean doNotTrustAllPolymorphic = false;
+
+    @Argument(
+            fullName = OMIT_ALLELES_IN_HDF5_LONG_NAME,
+            doc = "If true, omit alleles in output HDF5 files in order to decrease file sizes.",
+            optional = true
+    )
+    boolean omitAllelesInHDF5 = false;
+
+    @Argument(
+            fullName = DO_NOT_GZIP_VCF_OUTPUT_LONG_NAME,
+            doc = "If true, VCF output will not be compressed.",
+            optional = true
+    )
+    boolean doNotGZIPVCFOutput = false;
+
+    private final Set<String> ignoreInputFilterSet = new TreeSet<>();
+    Set<VariantType> variantTypesToExtract;
+    TreeSet<String> resourceLabels = new TreeSet<>();
+
+    File outputAnnotationsFile;
+    VariantContextWriter vcfWriter;
+
+    LabeledVariantAnnotationsData data;
+
+    @Override
+    public void onTraversalStart() {
+
+        ignoreInputFilterSet.addAll(ignoreInputFilters);
+
+        variantTypesToExtract = EnumSet.copyOf(variantTypesToExtractList);
+
+        outputAnnotationsFile = new File(outputPrefix + ANNOTATIONS_HDF5_SUFFIX);
+        final String vcfSuffix = doNotGZIPVCFOutput ? ".vcf" : ".vcf.gz";
+        final File outputVCFFile = new File(outputPrefix + vcfSuffix);
+
+        // TODO this validation method should perhaps be moved outside of the CNV code
+        CopyNumberArgumentValidationUtils.validateOutputFiles(outputAnnotationsFile, outputVCFFile);
+
+        for (final FeatureInput<VariantContext> resource : resources) {
+            final TreeSet<String> trackResourceLabels = resource.getTagAttributes().entrySet().stream()
+                    .filter(e -> e.getValue().equals("true"))
+                    .map(Map.Entry::getKey)
+                    .sorted()
+                    .collect(Collectors.toCollection(TreeSet::new));
+            resourceLabels.addAll(trackResourceLabels);
+            logger.info( String.format("Found %s track: labels = %s", resource.getName(), trackResourceLabels));
+        }
+        resourceLabels.forEach(String::intern);
```

Comment:
Hmm, I actually had a TODO here that I deleted just before opening the PR about checking whether this was necessary. Now I can't remember why I added it. Probably thought it might have an effect on the memory needed for per-site labels. Let me come back to this after doing some profiling!